eudemo/eudemo/maintenance/lower_port/designer.py

```diff
@@ -0,0 +1,595 @@
+                f"Making {self.params.lower_duct_angle.name} more negative ",
+                "or raising the z-point of the straight duct may help.",
+            )
+
+        # left-most
+        intc_point = min(intc_points, key=lambda p: p[0])
+        intc_point_inset = (
+            intc_point[0] - self.lower_duct_straight_x_offset,
+            intc_point[2],
+        )
+
+        x_duct_start = intc_point_inset[0]
+        z_duct_top = intc_point_inset[1]
+        z_duct_bottom = intc_point_inset[1] - self.lower_port_z_length
+
+        return (
+            make_polygon(
+                {
+                    "x": np.array(
+                        [
+                            x_duct_start,
+                            x_duct_extent,
+                            x_duct_extent,
+                            x_duct_start,
+                        ]
+                    ),
+                    "y": 0,
+                    "z": np.array(
+                        [
+                            z_duct_top,
+                            z_duct_top,
+                            z_duct_bottom,
+                            z_duct_bottom,
+                        ]
+                    ),
+                },
+                closed=True,
+            ),
+            # bottom left point
+            (x_duct_start, z_duct_bottom),
+        )
+
+    @staticmethod
+    def _xz_points_dist_away_from(
+        starting_xz_point: Union[Tuple, List],
+        gradient: float,
+        distance: float,
+    ) -> Tuple:
+        s_x = starting_xz_point[0]
+        s_z = starting_xz_point[1]
+        sqrt_value = np.sqrt(distance**2 / (1 + gradient**2))
+        f_x_pve = sqrt_value + s_x
+        f_z_pve = gradient * sqrt_value + s_z
+        f_x_nve = -sqrt_value + s_x
+        f_z_nve = gradient * -sqrt_value + s_z
+        return (
+            [f_x_nve, f_z_nve],
+            [f_x_pve, f_z_pve],
+        )
+
+    @staticmethod
+    def _make_xz_wire_from_points(
+        a_xz_point: Tuple,
+        b_xz_point: Tuple,
+    ) -> BluemiraWire:
+        return make_polygon(
+            {
+                "x": np.array([a_xz_point[0], b_xz_point[0]]),
+                "y": 0,
+                "z": np.array([a_xz_point[1], b_xz_point[1]]),
+            },
+        )
+
+    @staticmethod
+    def _intersection_points(
+        shape_a: BluemiraGeo,
+        shape_b: BluemiraGeo,
+    ) -> List[Tuple]:
+        dist, vects = distance_to(shape_a, shape_b)
+        if dist > 0.0001:  # not intersecting
+            return []
+        pois = []
+        for vect_pair in vects:
+            v = vect_pair[0]
+            pois.append((v[0], v[1], v[2]))
+        return pois
+
+    @staticmethod
+    def _closest_points(
+        shape_a: BluemiraGeo,
+        shape_b: BluemiraGeo,
+    ) -> List[Tuple]:
+        dist, vects = distance_to(shape_a, shape_b)
+        if dist < 0.0001:  # intersecting, return intersection points
```

Comment:
Same here

Reply:
also you are essentially doing this check twice, once here and the reverse in the function you call in the return. Is it possible to adjust to its only checked once?

Reply:
Sure no problem